### PR TITLE
xdr: changes for auth and removal of mutli-invoke (wip)

### PR DESCRIPTION
**Pull request by chowbao**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

https://github.com/stellar/go/issues/4895
Updates to support auth and removal of multi-invoke

* xdr changes up to https://github.com/stellar/stellar-xdr/commit/de264d81ea87dfaa19e089b60fed8eb0219b2ab4
* Core version 19.11.1-1328.cfa6a8918.focal~authchanges
* All instances of InvokeHostFunctions changed to InvokeHostFunction with only single HostFunction
* txnbuild updates
  * Added `Auth []xdr.SorobanAuthorizationEntry` to InvokeHostFunction struct
* Ingest
  * Replace `GetOperationEvents` with `GetDiagnosticEvents` to return diagnostic events
* Update Horizon operation_processor_tests for new InvokeHostFunction format
* sac_test updated to use ResultMetaXdr to get the InvokeHostFunction return ScVal

### Why

Updates needed for preview 10

### Known limitations

[TODO or N/A]

### CODE REVIEW

**Comment by chowbao** [comment]
@tsachiherman I pulled in the xdr changes needed to update the generated xdr.
Currently trying to get/build the new stellar-core image

File: services/horizon/internal/integration/contracts/Cargo.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ codegen-units = 1
 lto = true
 
 [workspace.dependencies.soroban-sdk]
-version = "0.8.0"
+# TODO: Does a new version need to be released to update to 8a7d2c3c8c5bc1664ab38f977ff947cc2a26d694
```
Comment (chowbao) [inline services/horizon/internal/integration/contracts/Cargo.toml:25]: @tsachiherman Unsure if there is any change needed to https://github.com/stellar/rs-soroban-sdk to update the version. 
From looking at this these changes are just used to generate wasm testdata

File: Makefile [hunk 2]
```diff
@@ -15,7 +15,7 @@ xdr/Stellar-internal.x \
 xdr/Stellar-contract-config-setting.x
 
 XDRGEN_COMMIT=80e38ef2a96489f6b501d4db3a350406e5aa3bab
-XDRNEXT_COMMIT=2f16687fdf6f4bcfb56805e2035f69997f4b34c4
+XDRNEXT_COMMIT=de264d81ea87dfaa19e089b60fed8eb0219b2ab4
```
[the remark this answers is not in the document]
Comment (chowbao) [reply Makefile:18]: Yeah I updated the `XDRNEXT_COMMIT` then ran `make xdr-update`. I guess copying the files would have worked too but I still would have had to run like `make xdr`

**Review by tsachiherman — APPROVED** [review]
rebasing to soroban-xdr-next-next so that we can keep working on this one while doing changes to soroban-rpc.

**Review by Dezzj — COMMENTED** [review]
Veiwed

**Pull request merged**